serde_derive/src/de.rs

```diff
@@ -2847,6 +2854,30 @@ fn field_i(i: usize) -> Ident {
     Ident::new(&format!("__field{}", i), Span::call_site())
 }
 
+fn wrap_stringify(
+    _params: &Parameters,
+    value_ty: &TokenStream,
+) -> (TokenStream, TokenStream) {
+    // TODO FIXME
+    let wrapper = quote! {
+        impl _serde::Deserialize for #value_ty {
+            fn deserialize<'de, __D>(__deserializer: __D) -> _serde::__private::Result<Self, __D::Error>
+            where
+                __D: _serde::Deserializer<'de>,
+                #value_ty: std::str::FromStr,
+                __D::Error: Sized + de::StdError,
+            {
+                let s = String::deserialize(__deserializer)?;
```

Comment:
This should not deserialize to a `String`. The code from PR #2017 has a better solution: https://github.com/serde-rs/serde/pull/2017/files#diff-2a062279ae10fa8cfea059371a1320afc87b998e7153017941e31ff7450ca64aR413-R426
This avoids an extra allocation and makes the code better for `no_std` environments.

Reply:
Thank you @jonasbb !